language/bytecode_verifier/src/type_memory_safety.rs

```diff
@@ -127,6 +139,113 @@ impl<'a> TypeAndMemorySafetyAnalysis<'a> {
         existing_borrows.is_empty()
     }
 
+    fn verify_instantiation(
+        env: (&[StructHandle], &[Kind], usize),
+        tys: &[SignatureToken],
+        type_formals: &[Kind],
```

Comment:
At this point I would call these constraints. Since we will be checking that for each `ty in tys` that the `ty` is a valid subkind